packages/cozy-procedures/src/components/Amount.jsx

```diff
@@ -1,22 +1,74 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import { withRouter } from 'react-router'
+import get from 'lodash/get'
 import Topbar from './Topbar'
-import { translate } from 'cozy-ui/transpiled/react/I18n'
+import {
+  Title,
+  translate,
+  Label,
+  Input,
+  Button
+} from 'cozy-ui/transpiled/react/'
+import creditApplicationTemplate from '../templates/creditApplicationTemplate'
 
 class Amount extends React.Component {
+  constructor(props) {
+    super(props)
+    this.confirm = this.confirm.bind(this)
+    this.inputRef = React.createRef()
+  }
+
   render() {
-    const { t } = this.props
+    const { t, amount, updateAmount } = this.props
+    const min = get(creditApplicationTemplate, 'procedureData.amount.min')
+    const max = get(creditApplicationTemplate, 'procedureData.amount.max')
+    const defaultValue = get(
+      creditApplicationTemplate,
+      'procedureData.amount.default'
+    )
     return (
       <div>
         <Topbar title={t('amount.title')} />
-        Amount
+        <Title>{t('amount.subtitle')}</Title>
+
+        <div>
+          <Label htmlFor="amount-input">{t('amount.label')}</Label>
+          <Input
+            type="number"
+            id="amount-input"
+            min={min}
+            max={max}
+            value={amount || defaultValue}
+            onChange={e => updateAmount(parseInt(e.target.value))}
+            inputRef={this.inputRef}
+          />
+        </div>
+
+        <div>
+          <Button
+            label={t('confirm')}
+            extension="full"
+            onClick={this.confirm}
+          />
+        </div>
       </div>
     )
   }
+
+  confirm() {
+    this.props.updateAmount(parseInt(this.inputRef.current.value))
```

Comment:
Since you already use have the value via onChange, can't you just remove this line ?

Reply:
It's trickier than that: on the first visit, a default value is set, but we don't want to save that default value — unless the explicitly clicks on the confirm button. It's the only reason why we need this other call to `updateAmount ` :(